src/formatting/items.rs

```diff
@@ -2405,15 +2407,13 @@ fn rewrite_fn_base(
                 Shape::indented(indent, context.config)
             } else {
                 let mut ret_shape = Shape::indented(indent, context.config);
-                if param_str.is_empty() {
-                    // Aligning with non-existent params looks silly.
+                if param_str.is_empty() || ret_on_nl {
+                    // Aligning with non-existent params looks silly, as does aligning the return
+                    // type when it is on a newline entirely disconnected from the parentheses of
+                    // the parameters.
                     force_new_line_for_brace = true;
-                    ret_shape = if context.use_block_indent() {
-                        ret_shape.offset_left(4).unwrap_or(ret_shape)
-                    } else {
-                        ret_shape.indent = ret_shape.indent + 4;
-                        ret_shape
-                    };
```

Comment:
#3731 added this, but I am not sure that the branch is needed. It doesn't break any tests, and I don't understand what `offset_left` is supposed to do in this case.

Reply:
Hmm. Would you mind leaving it in then, at least for now, so that I can go ahead and merge? I really appreciate @ahl test driving the 2.0 beta and would prefer to get the bug fix merged, and we can investigate the necessity of this conditional in a follow up.

Thoughts?

Reply:
Sounds good, I have reverted the changes.

Reply:
Awesome, thank you!